src/chrome/content/rules/Stack-Exchange.xml

```diff
@@ -1,177 +1,128 @@
+	<target host="www.stackoverflow.com" />
+		
+	<!-- Superuser -->
 	<target host="superuser.com" />
-	<target host="*.superuser.com" />
+	<target host="meta.superuser.com" />
+	<target host="www.superuser.com" />
+	
+	<!-- Misc -->
+	<target host="sstatic.net" />
+	<target host="cdn.sstatic.net" />
+	<target host="stackapps.com" />
 	<target host="teststackoverflow.com" />
-	<target host="*.teststackoverflow.com" />
-
+	<target host="www.teststackoverflow.com" />
 
-	<!--	Not secured by server:
-					-->
-	<!--securecookie host="^\.serverfault\.com$" name="^prov$" /-->
-	<!--securecookie host="^\.stackexchange\.com$" name="^prov$" /-->
-	<!--securecookie host="^stackoverflow\.com$" name="^rk-\d+$" /-->
-	<!--securecookie host="^\.stackoverflow\.com$" name="^prov$" /-->
+<!-- Securecookies -->
 
-	<securecookie host="^\.sstatic\.net$" name=".+" />
-	<securecookie host="\.stackexchange\.com$" name=".+" />
-	<securecookie host="^stackoverflow\.com$" name=".+" />
-	<securecookie host="^\.teststackoverflow\.com$" name=".+" />
+	<!-- Blogoverflow -->
+	<securecookie host="^blogoverflow\.com$" name=".+" />
 
+	<!-- Askubuntu -->
+	<securecookie host="^askubuntu\.com$" name=".+" />
+	<securecookie host="^meta\.askubuntu.com$" name=".+" />
 
-	<rule from="^http://(?:www\.)?blogoverflow\.com/$"
-		to="https://stackexchange.com/blogs" />
+	<!-- Mathoverflow -->
+	<securecookie host="^mathoverflow\.net$" name=".+" />
+	<securecookie host="^meta\.mathoverflow.net$" name=".+" />
 	
-	<rule from="^http://([^\/]+)\.blogoverflow\.com/"
-		to="https://$1.blogoverflow.com/" />
-
-	<rule from="^http://(?:(or\.)?cdn\.)?sstatic\.net/"
-		to="https://$1cdn.sstatic.net/" />
-
-	<rule from="^http://cdn-c(areers|hat)\.sstatic\.net/"
-		to="https://cdn-c$1.sstatic.net/" />
-
-	<rule from="^http://(?:www\.)?stackapps\.com/"
-		to="https://stackapps.com/" />
-
-	<rule from="^http://((?:\w+|discuss\.area51)\.)?stackexchange\.com/"
-		to="https://$1stackexchange.com/" />
-
-	<rule from="^http://((?:blog|www|meta|careers)\.)?stackoverflow\.com/"
-		to="https://$1stackoverflow.com/" />
-
-	<rule from="^http://app\.stacktack\.com/"
-		to="https://s3.amazonaws.com/stacktackapp/" />
-
-	<rule from="^http://(www\.|meta\.)?superuser\.com/"
-		to="https://$1superuser.com/" />
-
-	<rule from="^http://(\w\w)\.stackoverflow\.com/"
-		to="https://$1.stackoverflow.com/" />
-
-	<rule from="^http://meta\.(\w\w)\.stackoverflow\.com/"
-		to="https://meta.$1.stackoverflow.com/" />
+	<!-- Serverfault -->
+	<securecookie host="^clc\.serverfault\.com$" name=".+" />
+	<securecookie host="^meta\.serverfault\.com$" name=".+" />
+	<securecookie host="^serverfault\.com$" name=".+" />
+	
+	<!-- Stackauth -->
+	<securecookie host="^stackauth\.com$" name=".+" />
+	<securecookie host="^dev\.stackauth\.com$" name=".+" />
+	
+	<!-- Stackexchange -->
+	<securecookie host="^stackexchange\.com$" name=".+" />
+	<securecookie host="^([\w-]+)\.stackexchange\.com$" name=".+" />
```

Comment:
Okay, it looks like your fix was correct: Since `meta.*.stackexchange.com` does have a login capability, it probably relies on cookies. So the only cookies we can reliably secure are the ones set for a specific hostname, as you've got it set up. Have you checked the actual cookies set by the server to see if there are any of these, and if they are important?